src/extension/background-script/index.js

```diff
@@ -1,85 +1,40 @@
 import browser from "webextension-polyfill";
-import utils from "../../common/lib/utils";
-import Settings from "../../common/lib/settings";
-import Accounts from "../../common/lib/accounts";
-import connectors from "./connectors";
-
-import initLsatInterceptor from "./lsatInterceptor";
-
-let currentUnlockPassword; // TODO: rethink this
-let connector;
-let accounts = new Accounts();
-let settings = new Settings();
-const initConnector = async () => {
-  await accounts.load(); // load latest accounts from storage
-  const account = accounts.current;
-  console.log(account);
-  if (!account) {
-    connector = null;
-    console.log("Account not found");
-    return;
-  }
-  connector = new connectors[account.connector](account.config);
 
-  if (currentUnlockPassword) {
-    connector.unlock(currentUnlockPassword);
-  }
-
-  return connector.init();
-};
+import utils from "../../common/lib/utils";
 
-browser.storage.onChanged.addListener((changes) => {
-  // if the accounts change we initialize a new connector
-  // this also requires the user to unlock the account again
-  if (changes.accounts) {
-    initConnector();
-  }
-  // Update the general settings in the current connector settings
-  if (connector && changes.settings) {
-    connector.init();
-  }
-  // TODO: what to do with allowances?
-});
+import { router } from "./router";
+import state from "./state";
 
 const debugLogger = (message, sender) => {
-  if (settings && settings.debug) {
+  if (state.getState().settings.debug) {
     console.log("Background onMessage: ", message, sender);
   }
 };
 
 const handleInstalled = (details) => {
-  console.log("handle installed");
+  console.log(`Handle installed: ${details.reason}`);
   // TODO: maybe check if accounts are already configured?
   if (details.reason === "install") {
     utils.openUrl("welcome.html");
   }
 };
 
-// listen to calls from the content script and pass it on to the native application
+// listen to calls from the content script and calls the actions through the router
 // returns a promise to be handled in the content script
-const handleConnectorCalls = (message, sender, sendResponse) => {
+const routeCalls = (message, sender, sendResponse) => {
```

Comment:
Unused/unnecessary `sendResponse`. This was the old, callback-based way to return a message. Now the promise does it already.

```suggestion
const routeCalls = (message, sender) => {
```